sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -1396,6 +1396,17 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
       compressed = false,
       properties = rowStorage.properties ++ fileStorage.properties)
 
-    (ctx.LOCAL != null, storage, Some(DDLUtils.HIVE_PROVIDER))
+    val fileFormat = extractFileFormat(fileStorage.serde)
+    (ctx.LOCAL != null, storage, Some(fileFormat))
+  }
+
+  private def extractFileFormat(serde: Option[String]): String = {
+    if (serde.toString.contains("parquet")) {
```

Comment:
Although `None.toString` is `"None"`, this looks like a hack to me.

How about:
```
serde.map({ x =>
  val lowerCaseSerde = x.toLowerCase(Locale.ROOT)
  if (lowerCaseSerde.contains("parquet") "parquet"
  else if (lowerCaseSerde.contains("orc") "orc"
  else  DDLUtils.HIVE_PROVIDER
}).getOrElse( DDLUtils.HIVE_PROVIDER)
```